src/App/Cart/Service/CartCalculationService.php

```diff
@@ -23,13 +25,18 @@ public function calculateAllowedPackageTypes(PdkCart $cart): PackageTypeCollecti
         return PackageTypeCollection::fromAll()
             ->sortBySize(true)
             ->filter(function (PackageType $packageType) use ($cart) {
-                $lineHasPackageType = $cart->lines->containsStrict('product.settings.packageType', $packageType->name);
+                $packageTypeName    = $packageType->name;
+                $lineHasPackageType = $cart->lines->containsStrict('product.settings.packageType', $packageTypeName);
 
-                if ($lineHasPackageType && DeliveryOptions::PACKAGE_TYPE_MAILBOX_NAME === $packageType->name) {
+                if (! $this->isWeightUnderPackageTypeLimit($cart, $packageTypeName)) {
```

Comment:
Om hier een negated condition te vermijden, zou ik dan de functie zelf omdraaien.

En je gebruikt nu niet `$lineHasPackageType`, dus of die moet je wel gebruiken, of die assignment moet onder deze if statement.